src/windowing.jl

```diff
@@ -0,0 +1,61 @@
+struct WindowViewer{T<:Real, V<:AbstractVector{<:T}}
+    timeseries::V
+    halfwidth::Int
+    stride::Int
+    strided_indices::AbstractVector{Int}
+end
+
+"""
+
+    WindowViewer(
+        timeseries::V,
+        halfwidth::Int,
+        stride::Int,
+    ) where {T<:Real, V<:AbstractVector{<:T}}
+
+Returns struct of WindowViewer with fields:
+- `WindowViewer.timeseries`
+- `WindowViewer.halfwidth`
+- `WindowViewer.stride`
+- `WindowViewer.strided_indices`: a vector of indices at which the window viewing should be applied.
+"""
```

Comment:
This isn't the goal of documentation. The goal of documentation is to tell users how to **use** this thing, and what the input parameters mean. Here you are saying "make something with the input parameters" which doesn't help. 

You should say something like: "Initialize an iterator that generates views over the given `timeseries` based on a window with a given `halfwidth`, incrementing the window views with the given `stride`. You can use this directly with `map`, such as `map(std, WindoViewer(x, ...))` that would give you the moving-window-timeseries of the `std` of `x`. Alternatively, you can use it with the function `apply_over_windows`.

We have removed the bracketing information now, left, right, or whatever, but we can offer a simple function 

`vals, tvec = apply_over_windows(std, WindowViewer(x, ...), bracketing)` with `bracketing = "left", "right" or "center"` that makes the corresponding time vector so that it is easy to plot `vals` in the same plot as you plot `x`.

Reply:
Thanks, I'll keep this in mind for future!